### PR TITLE
Brave News feed fixes

### PR DESCRIPTION
**Pull request by petemill**
- Fix https://github.com/brave/brave-browser/issues/26277
  - Always set a featured item, even if the user follows no news-categorised sources
  - Score sources explicitly followed higher than sources shown via a channel. This will be very subtle and only really affect the first item from a source since recency has an exponential growth in score.

- Fix https://github.com/brave/brave-browser/issues/26303
  - When a user unfollows all sources and channels, remove all content from the feed without requiring a browser restart. Still requires page / content refresh.
  - Show a new no-content-configured card
  - Fix the error-getting-content card
  - Add buttons to both error cards for refresh or customize, depending on the error

image

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
On issues

### CODE REVIEW

**Comment by petemill** [comment]
Apologies for the combined PR, but the fixes are semi-related and the changes are somewhat minimal. Feel free to review the 3 commits independently.

File: components/brave_new_tab_ui/components/default/braveToday/cards/cardError.tsx [hunk 1]
```diff
@@ -38,20 +39,33 @@ function Icon () {
   )
 }
 
-export default function CardError () {
+interface MessageProps {
+  heading: string
+  message: string
+}
+export function UnidealMessageCard (props: MessageProps) {
```
Comment (fallaciousreasoning) [inline components/brave_new_tab_ui/components/default/braveToday/cards/cardError.tsx:46]: Big fan of the naming :+1:

File: components/brave_new_tab_ui/components/default/braveToday/cards/cardNoContent.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
```
Comment (fallaciousreasoning) [inline components/brave_new_tab_ui/components/default/braveToday/cards/cardNoContent.tsx:1]: Nit: 2022
```suggestion
// Copyright (c) 2022 The Brave Authors. All rights reserved.
```

File: components/resources/brave_news_strings.grdp [hunk 3]
```diff
@@ -69,6 +69,18 @@
   <message name="IDS_BRAVE_NEWS_SUGGESTIONS_SUBTITLE" desc="Subtitle of the suggested sources section">
     Some sources you might like. Suggestions are anonymous and matched only on your device.
   </message>
+  <message name="IDS_BRAVE_NEWS_ERROR_HEADING" desc="Message heading for when there is a Brave News error">
+    Oops…
+  </message>
+  <message name="IDS_BRAVE_NEWS_ERROR_MESSAGE" desc="Message body for when there is a Brave News error">
+    Brave News is experiencing some issues. Try again.
```
Comment (fallaciousreasoning) [inline components/resources/brave_news_strings.grdp:76]: Try again, sounds kinda brusque. Maybe `Please try again later`?

File: components/resources/brave_news_strings.grdp [hunk 4]
```diff
@@ -69,6 +69,18 @@
...
+  </message>
+  <message name="IDS_BRAVE_NEWS_NO_CONTENT_HEADING" desc="Message heading for when there is no Brave News content">
+    No content
+  </message>
+  <message name="IDS_BRAVE_NEWS_NO_CONTENT_MESSAGE" desc="Message body for when there is no Brave News content">
+    Brave News has no content to show. Go follow some sources!
```
Comment (fallaciousreasoning) [inline components/resources/brave_news_strings.grdp:82]: Completely & utterly optional: I wonder if it's worth having a button show on the card to open the Brave News dialog?

**Review by fallaciousreasoning — APPROVED** [review]
Only the nittiest of nits, LGTM (if you fix your license header for me :laughing:)

File: components/resources/brave_news_strings.grdp (hunk 3 above)
Comment (petemill) [reply components/resources/brave_news_strings.grdp:76]: Well, they might want to try immediately not later if it was a network issue (which it probably was) 😆 . Really we should have a button to refresh on this card and its text would be "Try again". But it's likely floating above this message somewhere.

File: components/brave_new_tab_ui/components/default/braveToday/cards/cardError.tsx (hunk 1 above)
Comment (petemill) [reply components/brave_new_tab_ui/components/default/braveToday/cards/cardError.tsx:53]: Very proud

File: components/brave_new_tab_ui/components/default/braveToday/cards/cardNoContent.tsx (hunk 2 above)
Comment (petemill) [reply components/brave_new_tab_ui/components/default/braveToday/cards/cardNoContent.tsx]: copy / paste fail! Fixed...

File: components/resources/brave_news_strings.grdp (hunk 3 above)
Comment (petemill) [reply components/resources/brave_news_strings.grdp]: Moved to a button with "Try Again" label

File: components/resources/brave_news_strings.grdp (hunk 4 above)
Comment (petemill) [reply components/resources/brave_news_strings.grdp]: Added the button
image

**Pull request merged**